### PR TITLE
[DO NOT MERGE] Added Audio Profile sample

### PR DESCRIPTION
**Pull request by happyhuman**

### CODE REVIEW

**Comment by happyhuman** [comment]
@nnegrey , @dizcology Can you please review this PR?

File: texttospeech/cloud-client/audio_profile.py [hunk 1]
```diff
@@ -0,0 +1,69 @@
+#!/usr/bin/env python
+
+# Copyright 2018 Google Inc. All Rights Reserved.
```
Comment (nnegrey) [inline texttospeech/cloud-client/audio_profile.py]: Google LLC

File: texttospeech/cloud-client/audio_profile_test.py [hunk 2]
```diff
@@ -0,0 +1,32 @@
+# Copyright 2018, Google, Inc.
```
Comment (nnegrey) [inline texttospeech/cloud-client/audio_profile_test.py]: Google LLC

File: texttospeech/cloud-client/audio_profile.py [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+"""Google Cloud Text-To-Speech API sample application for audio profile.
+
+Example usage:
+    python audio_profile.py --text "hello" --effects_profile_id
+        "telephony-class-application"
+"""
+
+import argparse
+
+from google.cloud import texttospeech_v1beta1 as texttospeech
```
Comment (nnegrey) [inline texttospeech/cloud-client/audio_profile.py]: This should be moved to right between line 31/32. (Just learned this myself)

File: texttospeech/cloud-client/audio_profile.py [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+
+
+# [START tts_synthesize_text]
```
Comment (nnegrey) [inline texttospeech/cloud-client/audio_profile.py]: update region tag

File: texttospeech/cloud-client/audio_profile.py [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+def synthesize_text(text, output, effects_profile_id):
```
Comment (nnegrey) [inline texttospeech/cloud-client/audio_profile.py]: def synthesize_text_with_audio_profile?

File: texttospeech/cloud-client/audio_profile.py [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+    """Synthesizes speech from the input string of text."""
+    client = texttospeech.TextToSpeechClient()
+
+    input_text = texttospeech.types.SynthesisInput(text=text)
```
Comment (nnegrey) [inline texttospeech/cloud-client/audio_profile.py]: Does this also support ssml? 

@dizcology Should there be a 2nd snippet for ssml support?

File: texttospeech/cloud-client/audio_profile_test.py [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+
+import audio_profile
+
+TEXT = 'hello'
+OUTOUT = 'output.mp3'
+EFFECTS_PROFILE_ID = 'telephony-class-application'
+
+
+def test_audio_profile(capsys):
+    if os.path.exists(OUTOUT):
```
Comment (nnegrey) [inline texttospeech/cloud-client/audio_profile_test.py]: We should probably do the clean-up after it runs.

File: texttospeech/cloud-client/audio_profile.py (hunk 1 above)
Comment (happyhuman) [reply texttospeech/cloud-client/audio_profile.py]: Good point, but it is also Google Inc in every other file.

File: texttospeech/cloud-client/audio_profile_test.py (hunk 7 above)
Comment (happyhuman) [reply texttospeech/cloud-client/audio_profile_test.py]: Done

File: texttospeech/cloud-client/audio_profile.py (hunk 1 above)
Comment (nnegrey) [reply texttospeech/cloud-client/audio_profile.py]: For new files, I've been told we make the effort to update them, but on older files we just leave them.

**Comment by happyhuman** [comment]
@dizcology , @nnegrey This PR passed the tests with no problem about a week or two before. All tests in texttospeech still pass locally on my computer. But now, the audio_profile_test.py fails in CicleCi. Have you guys encountered anything like this before? I have a feeling the build script is not installing the latest library and therefore the test fails.

File: texttospeech/cloud-client/audio_profile.py (hunk 1 above)
Comment (happyhuman) [reply texttospeech/cloud-client/audio_profile.py]: Done

File: texttospeech/cloud-client/audio_profile_test.py (hunk 2 above)
Comment (happyhuman) [reply texttospeech/cloud-client/audio_profile_test.py]: Done

**Comment by happyhuman** [comment]
@nnegrey , @dizcology when you have time, can you please finalize your review of this PR?

File: texttospeech/cloud-client/README.rst [hunk 8]
```diff
@@ -7,7 +7,7 @@ Google Cloud Text-to-Speech API Python Samples
    :target: https://console.cloud.google.com/cloudshell/open?git_repo=https://github.com/GoogleCloudPlatform/python-docs-samples&page=editor&open_in_editor=/README.rst
 
 
-This directory contains samples for Google Cloud Text-to-Speech API. The `Google Cloud Text-to-Speech API`_ enables you to generate and customize synthesized speech from text or SSML.
+This directory contains samples for Google Cloud Text-to-Speech API. The `Google Cloud Text To Speech API`_ enables you to generate and customize synthesized speech from text or SSML.
```
Comment (nnegrey) [inline texttospeech/cloud-client/README.rst]: Switch this one back I think.

File: texttospeech/cloud-client/audio_profile.py [hunk 9]
```diff
@@ -0,0 +1,70 @@
...
+
+    # Note: the voice can also be specified by name.
+    # Names of voices can be retrieved with client.list_voices().
+    voice = texttospeech.types.VoiceSelectionParams(language_code='en-US')
+
+    # Note: you can pass in multiple effects_profile_id. They will be applied
+    # in the same order they are provided.
+    audio_config = texttospeech.types.AudioConfig(
```
Comment (nnegrey) [inline texttospeech/cloud-client/audio_profile.py]: Nice to have a comment about what some of the effects_profile_ids are.

File: texttospeech/cloud-client/audio_profile.py [hunk 10]
```diff
@@ -0,0 +1,70 @@
...
+        effects_profile_id=[effects_profile_id])
+
+    response = client.synthesize_speech(input_text, voice, audio_config)
+
+    # The response's audio_content is binary.
+    with open(output, 'wb') as out:
+        out.write(response.audio_content)
+        print('Audio content written to file "%s"' % output)
+
+# [END tts_synthesize_text_with_audio_profile]
```
Comment (nnegrey) [inline texttospeech/cloud-client/audio_profile.py]: Could delete the new line.

File: texttospeech/cloud-client/audio_profile_test.py [hunk 11]
```diff
@@ -0,0 +1,34 @@
...
+    if os.path.exists(OUTPUT):
+        os.remove(OUTPUT)
+    assert not os.path.exists(OUTPUT)
+    audio_profile.synthesize_text_with_audio_profile(TEXT, OUTPUT,
+                                                     EFFECTS_PROFILE_ID)
+    out, err = capsys.readouterr()
+
+    assert ('Audio content written to file "%s"' % OUTPUT) in out
+    assert os.path.exists(OUTPUT)
+    os.remove(OUTPUT)
```
Comment (dizcology) [inline texttospeech/cloud-client/audio_profile_test.py]: Maybe use pytest's temp file for this.

**Pull request merged**